### PR TITLE
Rewrite autocomplete and close #535 too

### LINKED ISSUE #535: Allow multi-term autocomplete?
Given that the autocomplete service has all of the parameters of the search service (so it can work from the right set of terms), it could trivially generate a link from the term to the appropriate search to run.  This would be more REST/HATEOAS friendly than making the client construct the url by copying the match content into the q param of the search URL.

e.g. as per first half of:
  http://search.iiif.io/api/search/0.9/#response-1

### PR DESCRIPTION
**Pull request by azaroth42**
Sorry about the scope creep!

### CODE REVIEW

File: source/api/search/0.9/index.md [hunk 1]
```diff
@@ -577,10 +580,17 @@ http://example.org/service/identifier/autocomplete?q=bir&motivation=painting&box
 
 ### 4.2. Response
 
-The response is a list of very simple objects that include the term and its number of occurrences.  The number of terms provided is determined by the server.  The matching term is given as the value of the `match` property, and the number of occurences is the integer value of the `count` property.
-The term objects are of `@type` "search:Term", and this _MAY_ be included explicitly but it is not necessary.
+The response is a list (a "search:TermList") of very simple objects that include the term, a link to the search for that term, and the number of matches that search will have.  The number of terms provided in the list is determined by the server.
```
Comment (jpstroop) [inline source/api/search/0.9/index.md]: ~~very~~

**Comment by azaroth42** [comment]
:+1: to deleting very.

File: source/api/search/0.9/index.md [hunk 2]
```diff
@@ -593,15 +603,31 @@ The example request above might generate the following response:
...
+      "match": "birthday", 
+       "url": "http://example.org/service/identifier/search?motivation=painting&q=birthday",
+      "count": 21
+    }
   ]
 }
 {% endhighlight %}
 
-Semantic Tags or other URIs searchable via the `q` parameter might also have labels to display to the user, rather than the URI that matched.  The order of the terms might be by the label, rather than the matching URI, if there is an associated label.
+URIs or other data that are searchable via the `q` parameter might also have `label`s to display to the user, rather than the exact string that matched.  This can also be useful if stemming or other term normalization has occured, in order to display the original rather than the processed term.
```
Comment (jpstroop) [inline source/api/search/0.9/index.md]: "might also" :arrow_right:  "may also"? (sorry, it's a writing day, so my sensitivity is up)

Comment (azaroth42) [inline source/api/search/0.9/index.md]: I try to avoid lowercase may, should, must to avoid the RFC words.

Comment (jpstroop) [inline source/api/search/0.9/index.md]: I get it, but 'might' sounds wishy-washy to me. I think a normal 'may' is the correct word.

**Comment by azaroth42** [comment]
Rebased

**Comment by zimeon** [comment]
I made a couple of minor fixes and added a log entry for 0.9.2 in 1e2da6d, doesn't seem to show in at <http://635_multiterm.iiif.io/api/search/0.9/#autocomplete> yet. Otherwise :+1: from me

**Comment by azaroth42** [comment]
Yeah, we have capistrano working but not the webhooks yet.  Erin is working on getting Heaven installed to handle webhook and slack integration.

File: source/api/search/0.9/index.md [hunk 3]
```diff
@@ -556,7 +556,10 @@ The autocomplete service _MUST_ have an `@id` property with the value of the URI
...
 
-The other parameters (`motivation`, `date`,`user` and `box`), if supported, would refine the set of terms in the response to only ones from the annotations that match those filters.  For example, if the motivation is given as "painting", then only text from painting transcriptions will contribute to the list of terms in the response.
+The term should be parsed as a complete string, regardless of whether there is whitespace included in it. For example, the query term of "green bir" should not autocomplete on fields that match "green" and also include something that starts with "bir", but instead look for terms that start with the string "green bir".
+
+The other parameters (`motivation`, `date`, `user` and `box`), if supported, would refine the set of terms in the response to only ones from the annotations that match those filters.  For example, if the motivation is given as "painting", then only text from painting transcriptions will contribute to the list of terms in the response.
```
Comment (jpstroop) [inline source/api/search/0.9/index.md]: "if supported, ~~would~~ refine the set of terms"

**Comment by mikeapp** [comment]
:+1:

**Comment by jpstroop** [comment]
:+1:

**Comment by tomcrane** [comment]
:+1:

**Pull request merged**